### PR TITLE
Add backend conn stats JSON REST API and UI support

### PR DESCRIPTION
**Pull request by mihaitodor**
This adds the connection stats JSON REST API and the UI for visualising the data. Sample: https://media.giphy.com/media/xUPGcITQd0LHupfPYk/giphy.gif

### CODE REVIEW

File: web.go [hunk 1]
```diff
@@ -330,3 +336,78 @@ func expvarHandler(w http.ResponseWriter, r *http.Request) {
...
+					}
+				}
+			}
+		}
+	}
+
+	templatesRenderer.JSON(response, http.StatusOK, payload)
+}
+
+func getTotalConn(cl *connlimit.ConnLimiter) (int64, bool) {
```
Comment (mihaitodor) [inline web.go:394]: I am using these reflection hacks for now, but I'll talk with the project maintainers about customising their [oxy fork](https://github.com/containous/oxy) to expose the `totalConnections` field (and maybe add more stats). Not sure what they're planning to do with that fork, considering that the maintainers of [vulcand/oxy](https://github.com/vulcand/oxy) aren't very responsive.

**Comment by mihaitodor** [comment]
@relistan @bparli Please let me know if I can merge this one too.

File: web.go (hunk 1 above)
Comment (bparli) [reply web.go:394]: just FYI, I noticed a while back their vendored oxy is different than the main oxy so I think they're already doing some customization.

Comment (mihaitodor) [reply web.go:394]: Indeed, I can send them a PR for that lib too, but let's first see if they're interested in it and if they want any changes in the design. Before I send them the PR, I was hoping to get it running in our system to make sure everything is fine. There are loads of moving pieces...

**Comment by bparli** [comment]
LGTM :shipit:

**Review by bparli — APPROVED** [review]

**Pull request merged**